Content.Server/VoiceMask/VoiceMaskSystem.cs

```diff
@@ -1,113 +1,99 @@
-    {
-        if (message.Name.Length > HumanoidCharacterProfile.MaxNameLength || message.Name.Length <= 0)
-        {
-            _popupSystem.PopupEntity(Loc.GetString("voice-mask-popup-failure"), uid, message.Actor, PopupType.SmallCaution);
-            return;
-        }
-
-        component.VoiceName = message.Name;
-        _adminLogger.Add(LogType.Action, LogImpact.Medium, $"{ToPrettyString(message.Actor):player} set voice of {ToPrettyString(uid):mask}: {component.VoiceName}");
-
-        _popupSystem.PopupEntity(Loc.GetString("voice-mask-popup-success"), uid, message.Actor);
-
-        TrySetLastKnownName(uid, message.Name);
-
-        UpdateUI(uid, component);
+        args.Args.VoiceName = entity.Comp.VoiceMaskName;
+        args.Args.SpeechVerb = entity.Comp.VoiceMaskSpeechVerb ?? args.Args.SpeechVerb;
     }
 
+    #region User inputs from UI
     private void OnChangeVerb(Entity<VoiceMaskComponent> ent, ref VoiceMaskChangeVerbMessage msg)
     {
         if (msg.Verb is {} id && !_proto.HasIndex<SpeechVerbPrototype>(id))
             return;
 
-        ent.Comp.SpeechVerb = msg.Verb;
+        ent.Comp.VoiceMaskSpeechVerb = msg.Verb;
         // verb is only important to metagamers so no need to log as opposed to name
 
         _popupSystem.PopupEntity(Loc.GetString("voice-mask-popup-success"), ent, msg.Actor);
 
-        TrySetLastSpeechVerb(ent, msg.Verb);
-
         UpdateUI(ent, ent.Comp);
     }
 
-    private void OnSpeakerNameTransform(EntityUid uid, VoiceMaskComponent component, TransformSpeakerNameEvent args)
+    private void OnChangeName(EntityUid uid, VoiceMaskComponent component, VoiceMaskChangeNameMessage message)
     {
-        if (component.Enabled)
+        if (message.Name.Length > HumanoidCharacterProfile.MaxNameLength || message.Name.Length <= 0)
         {
-            /*
-            args.Name = _idCard.TryGetIdCard(uid, out var card) && !string.IsNullOrEmpty(card.FullName)
-                ? card.FullName
-                : Loc.GetString("voice-mask-unknown");
-                */
-
-            args.Name = component.VoiceName;
-            if (component.SpeechVerb != null)
-                args.SpeechVerb = component.SpeechVerb;
+            _popupSystem.PopupEntity(Loc.GetString("voice-mask-popup-failure"), uid, message.Actor, PopupType.SmallCaution);
+            return;
         }
+
+        component.VoiceMaskName = message.Name;
+        _adminLogger.Add(LogType.Action, LogImpact.Medium, $"{ToPrettyString(message.Actor):player} set voice of {ToPrettyString(uid):mask}: {component.VoiceMaskName}");
+
+        _popupSystem.PopupEntity(Loc.GetString("voice-mask-popup-success"), uid, message.Actor);
+
+        UpdateUI(uid, component);
     }
+    #endregion
 
-    private void OnMaskToggled(Entity<VoiceMaskComponent> ent, ref WearerMaskToggledEvent args)
+    #region UI
+    private void OnEquip(EntityUid uid, VoiceMaskComponent component, ClothingGotEquippedEvent args)
```

Comment:
I think this was done on purpose! If you do it with the GetItemActionsEvent then It'll allow you to change the voice mask settings when your holding the item. From what I remember, this wasn't wanted because then its super easy for security to metagame it without being called out (They have to equip it). The chameleon works the same way and only shows when its equipped.

It does seems suspect though so if you think there is a better way of doing it let me know!
(To be clear I also didn't even think about the event being called. I just left it how it was originally 😆)